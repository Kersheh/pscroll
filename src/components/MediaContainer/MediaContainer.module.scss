.mediaContainer {
  .media {
    box-sizing: border-box;
    max-width: calc(100vw / 5);

    @media only screen and (max-width: 2560px) {
      max-width: calc(100vw / 4);
    }

    @media only screen and (max-width: 1920px) {
      max-width: calc(100vw / 3);
    }

    @media only screen and (max-width: 1280px) {
      max-width: calc(100vw / 2);
    }

    @media only screen and (max-width: 640px) {
      max-width: calc(100vw / 1);
    }
  }
}