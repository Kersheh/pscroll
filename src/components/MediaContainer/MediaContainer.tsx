import { useState } from 'react';

import { isImage, isVideo } from 'src/utils';
import Spinner from 'src/components/Spinner/Spinner';
import styles from './MediaContainer.module.scss';

interface MediaContainerProps {
  src: string;
}
const MediaContainer = ({ src }: MediaContainerProps) => {
  const [loading, setLoading] = useState(true);

  return (
    <div className={styles.mediaContainer}>
      {isImage(src) && (
        <img
          style={loading ? { display: 'none' } : {}}
          onLoad={() => setLoading(false)}
          alt={src}
          src={src}
          className={styles.media}
          data-testid="img"
        />
      )}

      {isVideo(src) && (
        <video
          className={styles.media}
          style={loading ? { display: 'none' } : {}}
          onLoadedData={() => setLoading(false)}
          preload="metadata"
          autoPlay
          muted
          loop
          playsInline
          data-testid="video"
        >
          <source src={src} type="video/mp4" />
        </video>
      )}

      {loading && <Spinner color="white" />}
    </div>
  );
};

export default MediaContainer;
