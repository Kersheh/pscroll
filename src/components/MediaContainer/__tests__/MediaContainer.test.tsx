import React from 'react';
import { fireEvent, render, screen } from '@testing-library/react';
import MediaContainer from '../MediaContainer';

describe('<MediaContainer />', () => {
  it('should render image', () => {
    const { container } = render(<MediaContainer src="example.png" />);

    fireEvent.load(screen.getByTestId('img'));
    expect(container).toMatchSnapshot();
  });

  it('should render video', async () => {
    const { container } = render(<MediaContainer src="example.mp4" />);

    fireEvent.loadedData(screen.getByTestId('video'));
    expect(container).toMatchSnapshot();
  });

  it('should display spinner and hide image while loading', () => {
    render(<MediaContainer src="example.png" />);

    expect(screen.getByTestId('spinner')).toBeInTheDocument();
    expect(screen.getByTestId('img')).not.toBeVisible();

    fireEvent.load(screen.getByTestId('img'));
    expect(screen.queryByTestId('spinner')).not.toBeInTheDocument();
    expect(screen.getByTestId('img')).toBeVisible();
  });

  it('should display spinner and hide video while loading', () => {
    render(<MediaContainer src="example.mp4" />);

    expect(screen.getByTestId('spinner')).toBeInTheDocument();
    expect(screen.getByTestId('video')).not.toBeVisible();

    fireEvent.loadedData(screen.getByTestId('video'));
    expect(screen.queryByTestId('spinner')).not.toBeInTheDocument();
    expect(screen.getByTestId('video')).toBeVisible();
  });
});
