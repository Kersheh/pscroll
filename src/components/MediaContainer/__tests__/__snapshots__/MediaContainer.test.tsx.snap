// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<MediaContainer /> should render image 1`] = `
<div>
  <div
    class="mediaContainer"
  >
    <img
      alt="example.png"
      class="media"
      data-testid="img"
      src="example.png"
      style=""
    />
  </div>
</div>
`;

exports[`<MediaContainer /> should render video 1`] = `
<div>
  <div
    class="mediaContainer"
  >
    <video
      autoplay=""
      class="media"
      data-testid="video"
      loop=""
      playsinline=""
      preload="metadata"
      style=""
    >
      <source
        src="example.mp4"
        type="video/mp4"
      />
    </video>
  </div>
</div>
`;
